src/client/Microsoft.Identity.Client/ManagedIdentity/AbstractManagedIdentity.cs

```diff
@@ -293,5 +295,54 @@ private static void CreateAndThrowException(string errorCode,
 
             throw exception;
         }
+
+        /// <summary>
+        /// Sets the claims and capabilities in the request.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="parameters"></param>
+        protected virtual void ApplyClaimsAndCapabilities(
+            ManagedIdentityRequest request,
+            AcquireTokenForManagedIdentityParameters parameters)
+        {
+            IEnumerable<string> clientCapabilities = _requestContext.ServiceBundle.Config.ClientCapabilities;
+
+            // If claims are present, set bypass_cache=true
+            if (!string.IsNullOrEmpty(parameters.Claims))
+            {
+                SetRequestParameter(request, "bypass_cache", "true");
```

Comment:
We didn't agree to this?

Reply:
@bgavrilMS We will finalize this after today's meeting with another MIRP. Basically we are trying to see if the proposed solution fits for all MIRPs.

Reply:
We will remove `bypass_cache` and pass the bad token hash for all MSI v1 sources